src/Core/Framework/Adapter/Cache/CacheCompilerPass.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Adapter\Cache;
+
+use Shopware\Core\Framework\Adapter\AdapterException;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+#[Package('framework')]
+class CacheCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        $storage = $container->getParameter('shopware.cache.invalidation.delay_options.storage');
+
+        switch ($storage) {
+            case 'mysql':
```

Comment:
do we need constants or an enum for the storage type? or is this the only place? and no other endpoints will be added most likely?